wgpu-native/src/lib.rs

```diff
@@ -168,46 +168,100 @@ pub struct ByteArray {
     pub length: usize,
 }
 
-pub type InstanceId = hub::Id;
+macro_rules! transparent {
+    ($i:item) => (
+        #[repr(transparent)]
+        #[derive(Clone, Copy, Debug, Hash, PartialEq)]
+        #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+        $i
+    )
+}
+
+use hub::{Index, Epoch, NewId, Id};
+macro_rules! new_id {
```

Comment:
why can't we have a single macro instead of 2?

Reply:
I agree, I was just adding a new macro since I needed a new one, but we can merge both into one